iocore/cache/Cache.cc

```diff
@@ -2000,6 +2000,12 @@ CacheProcessor::mark_storage_offline(CacheDisk *d ///< Target disk
   uint64_t total_dir_delete   = 0;
   uint64_t used_dir_delete    = 0;
 
+  /* Don't mark it again, it will invalidate the stats! */
+  if (d->offline) {
+    return this->has_online_storage();
+  }
+  d->offline = true;
```

Comment:
Why do yo introduce a new flag rather than making the code conditional on the ``DISK_BAD`` check? e.g.
```C
if (!DISK_BAD(d)) {
  SET_DISK_BAD(d);

  // Do all the other stuff ...
}
```